apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/class-wp-rest-wpcom-block-editor-first-post-published-modal-controller.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * WP_REST_WPCOM_Block_Editor_First_Post_Published_Modal_Controller file.
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE;
+
+/**
+ * Class WP_REST_WPCOM_Block_Editor_First_Post_Published_Modal_Controller.
+ */
+class WP_REST_WPCOM_Block_Editor_First_Post_Published_Modal_Controller extends \WP_REST_Controller {
+	/**
+	 * WP_REST_WPCOM_Block_Editor_First_Post_Published_Modal_Controller constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'block-editor/should-show-first-post-published-modal';
+	}
+
+	/**
+	 * Register available routes.
+	 */
+	public function register_rest_route() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'should_show_first_post_published_modal' ),
+					'permission_callback' => array( $this, 'permission_callback' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Callback to determine whether the request can proceed.
+	 *
+	 * @return boolean
+	 */
+	public function permission_callback() {
+		return current_user_can( 'read' );
+	}
+
+	/**
+	 * Should we show the first post published modal
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function should_show_first_post_published_modal() {
+		// As we has synced the `has_never_published_post` option to part of atomic sites but we cannot
+		// update the value now, always return false to avoid showing the modal at every publishing until
+		// we can update the value on atomic sites.
```

Comment:
Maybe link to the diff to make it more clear what you're referring to 

```suggestion
		// we can update the value on atomic sites. See D69932-code
```